contrib/adsk/libraries/adsklib/adsklib_legacy_defs.mtlx

```diff
@@ -95,6 +95,41 @@
     <output name="out" type="float" default="0.0" />
   </nodedef>
 
+  <nodedef name="ND_util_specklenoise_float" node="util_specklenoise" nodegroup="adsk_legacy">
+    <input name="position" type="vector3" defaultgeomprop="Pobject" />
+    <output name="out" type="float" default="0.0" />
+  </nodedef>
+
+  <nodedef name="ND_legacy_speckle_color3" node="legacy_speckle_color3" nodegroup="adsk_legacy" version="1.0.1" isdefaultversion="true" >
+    <input name="color1" type="color3" value="1, 1, 1" uisoftmin="0,0,0" uisoftmax="1,1,1" />
+    <input name="color2" type="color3" value="0, 0, 0" uisoftmin="0,0,0" uisoftmax="1,1,1" />
+    <input name="position" type="vector3" defaultgeomprop="Pobject" />
+    <input name="size" type="float" value="1" uisoftmin="0.0" uisoftmax="10.0" />
+    <input name="octaves" type="float" value="4.0" uisoftmin="0.0" uisoftmax="8.0" />
+    <input name="offset_x" type="float" value="0" uimin="0.0" uisoftmax="10.0" unittype="distance" />
+    <input name="offset_y" type="float" value="0" uimin="0.0" uisoftmax="10.0" unittype="distance" />
+    <input name="offset_z" type="float" value="0" uimin="0.0" uisoftmax="10.0" unittype="distance" />
+    <input name="rotate_x" type="float" value="0" uimin="0.0" uisoftmax="360.0" unittype="angle" />
+    <input name="rotate_y" type="float" value="0" uimin="0.0" uisoftmax="360.0" unittype="angle" />
+    <input name="rotate_z" type="float" value="0" uimin="0.0" uisoftmax="360.0" unittype="angle" />
+    <output name="out" type="color3" />
+  </nodedef>
+
+  <nodedef name="ND_legacy_marble_color3" node="legacy_marble_color3" nodegroup="adsk_legacy" version="1.0.1" isdefaultversion="true" >
+    <input name="position" type="vector3" defaultgeomprop="Pobject" />
+    <input name="size" type="float" value="0.7636" uimin="0.0" uisoftmax="10.0" />
+    <input name="width" type="float" value="0.070283" uimin="0.0" uisoftmax="10.0" />
+    <input name="stone_color" type="color3" value="0.0303, 0.0321, 0.0122" uisoftmin="0,0,0" uisoftmax="1,1,1" />
+    <input name="vein_color" type="color3" value="0.7832, 0.7832, 0.5755" uisoftmin="0,0,0" uisoftmax="1,1,1" />
+    <input name="offset_x" type="float" value="0" uimin="0.0" uisoftmax="10.0" unittype="distance" />
+    <input name="offset_y" type="float" value="0" uimin="0.0" uisoftmax="10.0" unittype="distance" />
+    <input name="offset_z" type="float" value="0" uimin="0.0" uisoftmax="10.0" unittype="distance" />
+    <input name="rotate_x" type="float" value="0" uimin="0.0" uisoftmax="360.0" unittype="angle" />
+    <input name="rotate_y" type="float" value="0" uimin="0.0" uisoftmax="360.0" unittype="angle" />
+    <input name="rotate_z" type="float" value="0" uimin="0.0" uisoftmax="360.0" unittype="angle" />
```

Comment:
Can we make these `vector3` offset and `vector3` rotation? It will reduce the number of uniforms, I guess we can't extract them for fractal later?

Reply:
I matched the UI inputs, where we have 6 separate fields, 3 offsets and 3 rotations.
I'd like to revise one day, but I lack the knowledge. In RRT the transforms are a 4x4 matrix. Not sure where that comes from.
It's possible that when we will work on the converter for these, we realize the inputs are different. For now this is good for graph testing.